src/components/core/PivotTable.tsx

```diff
@@ -57,12 +87,50 @@ export type IGetPage = (
     offset: number[]
 ) => Promise<Execution.IExecutionResponses | null>;
 
-export type IOnSuccess = (execution: Execution.IExecutionResponses, columnDefs: IGridHeader[]) => void;
+export type IOnSuccess = (execution: Execution.IExecutionResponses, columnDefs: ColDef[]) => void;
+
+export const getDrillRowData = (leafColumnDefs: ColDef[], rowData: {[key: string]: any}) => {
+    return leafColumnDefs.reduce((drillRow, colDef: ColDef) => {
+        const { type } = colDef;
+        // colDef without field is a utility column (e.g. top column label)
+        if (colDef.field) {
+            if (type === MEASURE_COLUMN) {
+                return [...drillRow, rowData[colDef.field]];
+            }
+            const drillItem = rowData.drillItemMap[colDef.field];
+            if (drillItem && (type === COLUMN_ATTRIBUTE_COLUMN || type === ROW_ATTRIBUTE_COLUMN)) {
+                return [...drillRow, {
+                    id: drillItem.uri.split('?id=')[1],
+                    title: rowData[colDef.field]
+                }];
+            }
+        }
+        return drillRow;
+    }, []);
+};
+
+export const getTreeLeaves = (tree: any, getChildren = (node: any) => node && node.children) => {
+    const leaves = [];
+    const nodes = Array.isArray(tree) ? [...tree] : [tree];
+    let node;
+    let children;
+    while (
+        // tslint:disable-next-line:no-conditional-assignment ban-comma-operator
```

Comment:
so we actually ban comma operator, we ban conditional assignment, but you have decided to use in anyway? 😄

Reply:
Yes, this is a good place to use comma operator. Comma operator is part of ES language and is banned because people usually use it by mistake. This is not a mistake, this is why we have tslint:disable-next-line.
https://stackoverflow.com/questions/9579546/when-is-the-comma-operator-useful